File: src/main/java/org/mcphoton/impl/command/StopCommand.java
/*
 * Copyright (c) 2016 MCPhoton <http://mcphoton.org> and contributors.
 *
 * This file is part of the Photon Server Implementation <https://github.com/mcphoton/Photon-Server>.
 *
 * The Photon Server Implementation is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * The Photon Server Implementation is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
package org.mcphoton.impl.command;

import org.mcphoton.command.Command;
import org.mcphoton.messaging.Messageable;

/**
 *
 * @author TheElectronWill
 */
public class StopCommand implements Command {

	@Override
	public void execute(Messageable source, String[] args) {
		System.exit(0);
	}

	@Override
	public String getName() {
		return "stop";
	}

	@Override
	public String getDescription() {
		return "Stop the server. Only for opped players";
	}

}

File: core/src/main/scala/org/mcphoton/command/StopCommand.scala
package org.mcphoton.command

import org.mcphoton.messaging.ChatMessageable
import org.mcphoton.runtime.ExecutionGroup
import org.mcphoton.server.PhotonServer
import org.mcphoton.world.World

/**
 * @author TheElectronWill
 */
class StopCommand extends Command("stop") {
	override def execute(sender: ChatMessageable, args: Seq[String])
						(implicit w: World, exgroup: ExecutionGroup): Unit = {
		PhotonServer.shutdown()
	}
}
